app-frontend/src/app/pages/market/search/search.controller.js

```diff
@@ -1,126 +1,156 @@
+/* global _ */
+
 export default class MarketSearchController {
     constructor( // eslint-disable-line max-params
-        $log, $state
+        $log, $state, toolService, toolTagService, toolCategoryService
     ) {
         'ngInject';
+        this.toolService = toolService;
+        this.toolTagService = toolTagService;
+        this.toolCategoryService = toolCategoryService;
         this.$state = $state;
         this.$log = $log;
+        this.nonSpaceDelimiter = '|';
```

Comment:
Setting the non space delimiter here to accommodate tags and categories that contain spaces